kyo-prelude/shared/src/main/scala/kyo/Local.scala

```diff
@@ -57,28 +55,68 @@ abstract class Local[A]:
       * @return
       *   The result of running the effect with the updated value
       */
-    final def update[B, S](f: A => A)(v: B < S)(using Frame): B < S =
-        ContextEffect.handle(
-            Tag[Local.State],
-            Map(this -> f(default)),
-            map => map.updated(this, f(map.getOrElse(this, default).asInstanceOf[A]).asInstanceOf[AnyRef])
-        )(v)
+    def update[B, S](f: A => A)(v: B < S)(using Frame): B < S
 end Local
 
 /** Companion object for Local, providing utility methods for creating Local instances. */
 object Local:
 
-    /** Creates a new Local instance with the given default value.
+    /** Creates a new regular Local instance with the given default value.
+      *
+      * Regular locals are similar to inheritable thread locals, where child fibers inherit the value from their parent fiber.
       *
       * @param defaultValue
       *   The default value for the Local
       * @return
-      *   A new Local instance
+      *   A new regular Local instance
       */
     @nowarn("msg=anonymous")
     inline def init[A](inline defaultValue: A): Local[A] =
-        new Local[A]:
+        new Base[A, State]:
+            def tag             = Tag[State]
             lazy val default: A = defaultValue
 
-    sealed private trait State extends ContextEffect[Map[Local[?], AnyRef]]
+    /** Creates a new isolated Local instance with the given default value.
+      *
+      * Isolated locals are similar to non-inheritable thread locals, where child fibers always start with the default value and do not
+      * inherit from their parent fiber.
+      *
+      * @param defaultValue
+      *   The default value for the Local
+      * @return
+      *   A new isolated Local instance
+      */
+    @nowarn("msg=anonymous")
+    inline def initIsolated[A](inline defaultValue: A): Local[A] =
+        new Base[A, IsolatedState]:
+            def tag             = Tag[IsolatedState]
+            lazy val default: A = defaultValue
+
+    object internal:
+
+        sealed private[kyo] trait State         extends ContextEffect[Map[Local[?], AnyRef]]
+        sealed private[kyo] trait IsolatedState extends ContextEffect[Map[Local[?], AnyRef]] with ContextEffect.Isolated
```

Comment:
Why does this need to be a `trait`?

Reply:
All effect types are encoded as `sealed trait` without any sub-classes since they aren't instantiated, being only used for matching tags. It could be `sealed abstract class` instead.